annotationframeworkclient/materializationengine.py

```diff
@@ -644,46 +645,62 @@ def map_filters(self, filters, timestamp, timestamp_past):
         Returns:
             [type]: [description]
         """
+        timestamp = pytz.UTC.localize(timestamp)
+
         new_filters = []
-        vals = []
+        root_ids = []
         for filter_dict in filters:
             if filter_dict is not None:
                 for col, val in filter_dict.items():
                     if col.endswith("root_id"):
                         if not isinstance(val, (Iterable, np.ndarray)):
-                            vals.append([val])
+                            root_ids.append([val])
                         else:
-                            vals.append(val)
+                            root_ids.append(val)
 
         # if they are all None then we can safely return now
-        if len(vals) == 0:
+        if len(root_ids) == 0:
             return [None, None, None], {}
-        vals = np.unique(np.concatenate(vals))
-        filter_vals_latest = self.cg_client.is_latest_roots(vals, timestamp=timestamp)
-        if not np.all(filter_vals_latest):
-            not_latest = vals[~filter_vals_latest]
+        root_ids = np.unique(np.concatenate(root_ids))
+
+        filter_timed_end = self.cg_client.is_latest_roots(root_ids, timestamp=timestamp)
+        filter_timed_start = self.cg_client.get_root_timestamps(root_ids) < timestamp
```

Comment:
if this isn't implemented in Minnie we can't merge and release it to pypi till it is.

Reply:
just added that to minnie and deployed